File: Hackerank/Days 12/main.js
'use strict';

class Person {
    constructor(firstName, lastName, idNumber) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.idNumber = idNumber;
    }

    printPerson() {
        console.log(`Name: ${this.lastName}, ${this.firstName}`);
        console.log(`ID: ${this.idNumber}`);
    }
}

class Student extends Person {
    constructor(firstName, lastName, idNumber, scores) {
        super(firstName, lastName, idNumber);
        this.scores = scores;
    }

    calculate() {
        let sum = this.scores.reduce((a, b) => a + b, 0);
        let avg = sum / this.scores.length;

        if (avg >= 90 && avg <= 100) return 'O';
        if (avg >= 80 && avg < 90) return 'E';
        if (avg >= 70 && avg < 80) return 'A';
        if (avg >= 55 && avg < 70) return 'P';
        if (avg >= 40 && avg < 55) return 'D';
        return 'T';
    }
}

const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf-8').trim().split('\n');

let line = input[0].split(' ');
let firstName = line[0];
let lastName = line[1];
let id = line[2];
let scores = input[2].split(' ').map(Number);

let s = new Student(firstName, lastName, id, scores);
s.printPerson();
console.log("Grade: " + s.calculate());

File: Hackerank/Days 12/main.py
class Person:
    def __init__(self, firstName, lastName, idNumber):
        self.firstName = firstName
        self.lastName = lastName
        self.idNumber = idNumber
    def printPerson(self):
        print("Name:", self.lastName + ",", self.firstName)
        print("ID:", self.idNumber)

class Student(Person):
    def __init__(self, firstName, lastName, idNumber, scores):
        super().__init__(firstName, lastName, idNumber)
        self.scores = scores

    def calculate(self):
        promedio = sum(self.scores) / len(self.scores)
        
        if 90 <= promedio <= 100:
            return 'O'
        elif 80 <= promedio < 90:
            return 'E'
        elif 70 <= promedio < 80:
            return 'A'
        elif 55 <= promedio < 70:
            return 'P'
        elif 40 <= promedio < 55:
            return 'D'
        else:
            return 'T'

line = input().split()
firstName = line[0]
lastName = line[1]
idNum = line[2]
numScores = int(input()) # not used, but read from input
scores = list(map(int, input().split()))

s = Student(firstName, lastName, idNum, scores)
s.printPerson()
print("Grade:", s.calculate())